secp256k1.go

```diff
@@ -110,6 +114,19 @@ func DeserializePrivateKey(data *SerializedPrivateKey) (key *PrivateKey, err err
 	return &PrivateKey{*data}, nil
 }
 
+// DeserializePrivateKeyFromSlice returns a PrivateKey type from a 32 byte private key slice.
+// will verify that it's 32 byte and it's a valid private key(Group Order > key > 0)
+func DeserializePrivateKeyFromSlice(data []byte) (key *PrivateKey, err error) {
+	if len(data) != SerializedPrivateKeySize {
+		return nil, errors.Errorf("invalid private key length got %d, expected %d", len(data),
+			SerializedPrivateKeySize)
+	}
+
+	serializedKey := &SerializedPrivateKey{}
+	copy(serializedKey[:], data)
```

Comment:
I might be able to get away with skipping this copy by creating a third internal function, but I think it's Go and I shouldn't overthink one copy

Reply:
Yeah, it's a constant-bound copy of two-digit bytes, better keep it as is to avoid complexity, unless we see it creates bottlenecks.